### PR TITLE
Adding Lambda layers documentation

### PR DESCRIPTION
**Pull request by brene**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/function/layers.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+---
+title: Reuse code & assets using layers
+description: Use Amplify CLI's Lambda layer capability to reuse code & assets across functions.
+---
+
+Lambda layers allows you to pull common code & assets for your Lambda function into a centralized location. With Lambda layers you can:
```
Comment (akshbhu) [inline docs/cli/function/layers.md:6]: allow

File: docs/cli/function/layers.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+```
+
+```console 
+? Provide a name for your Lambda layer: (layer-name)
+? Select up to 2 compatible runtimes:
+o NodeJS
+o Python
+```
+
+Next, you'll be guided through a workflow to provide a **layer name**, select **supported runtimes**. Currently, Amplify CLI provides NodeJS, and Python runtime support for layers.
```
Comment (edwardfoyle) [inline docs/cli/function/layers.md:42]: and select

File: docs/cli/function/layers.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+> Specific AWS organization
+> Public (everyone on AWS can use this layer)
+```
+
+After that, you'll be asked to configure your **layer's permission**.
+
+The **current AWS account will always have access to this layer**.
+In addition, the customer can configure access for:
+- **Specific AWS accounts**: provide a comma-separated list of AWS Account IDs to provide access to them.
+- **Specific AWS organizations**: provide a comma-separated list of AWS Organization IDs to provide access to them. *AWS Organization IDs start with `o-`.*
```
Comment (edwardfoyle) [inline docs/cli/function/layers.md:58]: extraneous * at the end of the line

File: docs/cli/function/layers.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+A `python` folder is auto-generated for you. In there you'll find a skeleton folder structure to add your Python packages to. Add your packages to `lib/python3.8/site-packages` folder and then you can use it from within your Python function same as any other `import`. 
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+### Add shared assets
+
+Any assets like large images or other files that you want to share across various functions can be placed in the `amplify/backend/function/<layer-name>/opt/` folder. Your function's code can import any assets by looking for files in the `opt/` path.
```
Comment (edwardfoyle) [inline docs/cli/function/layers.md:107]: should they look in /opt/ (opt within the root folder) or is it something like PATH/opt?

File: docs/cli/function/layers.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+o NodeJS
+o Python
+```
+
+Next, you'll be guided through a workflow to provide a **layer name**, select **supported runtimes**. Currently, Amplify CLI provides NodeJS, and Python runtime support for layers.
+
+TODO UPDATE TEXT BASED ON USER FEEDBACK
+```console
+? Who should have permission to use this layer?
+> Only the current AWS account
```
Comment (akshbhu) [inline docs/cli/function/layers.md:47]: we have removed this option

File: docs/cli/function/layers.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+Content changes in Lambda layer {LAYER_NAME} detected. What permissions do you want to grant to this new layer version?
+> The same permission as the latest layer version
+> Only accessible by the current account. You can always edit this later with: amplify update function
+```
+
+When a layer's content has been modified, Amplify CLI prompts you if the functions dependent on the layer should update to the latest layer version.
+
+```console
+You've updated Lambda layers: layer1, layer2...
+Do you want functions using the layers to upgrade to the latest version? (y/N)
```
Comment (edwardfoyle) [inline docs/cli/function/layers.md:163]: if someone runs amplify push --yes we need to select "no" here

File: docs/cli/function/layers.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+Now, you've successfully added a layer to your function.
+
+## Deploy Lambda layers & functions with Lambda layers
+
+Once you're ready with your changes in your layer and functions, you can deploy them by running `amplify push`. 
+
+If a Layer’s content has been updated and it has permissions associated, Amplify CLI will prompt you whether you want to carry the permissions forward to a newer version.
+
+```console
+Content changes in Lambda layer {LAYER_NAME} detected. What permissions do you want to grant to this new layer version?
```
Comment (jhockett) [inline docs/cli/function/layers.md:154]: We need a standard way of representing a layer name.

"\<lambda-layer-name\>", `layer-name`, {LAYER_NAME}

File: docs/cli/function/layers.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+```console
+You've updated Lambda layers: layer1, layer2...
+Do you want functions using the layers to upgrade to the latest version? (y/N)
+```
+
+## Update layer content
+
+Any file changes within a layer's folder are automatically tracked by Amplify CLI. If there are changes available, the Amplify CLI will create a new layer version with the changes and ask during `amplify push` if you'd like to update all dependent functions to use the latest version.
+
+You also have the ability to manually increment the layer version is added to a function by running `amplify update function` and selecting the `function` that you want to change the layer configuration for.
```
Comment (akshbhu) [inline docs/cli/function/layers.md:170]: this is confusing

Comment (jhockett) [inline docs/cli/function/layers.md:170]: This is not entirely true. A layer's version is only updated if runtimes change, or if content changes are detected in the layer folder

**Pull request merged**